infrastructure/kubernetes/modules/storage/main.tf

```diff
@@ -0,0 +1,11 @@
+resource "kubernetes_storage_class" "azurefile_csi_nfs" {
+  metadata {
+    name = var.backend_storage_class
+  }
+  storage_provisioner = "file.csi.azure.com"
+  reclaim_policy      = "Delete"
+  parameters = {
+    protocol = "nfs"
```

Comment:
`azurefile_csi` (based on SMB rather than NFS) comes included with AKS clusters, but
- it is case-preserving, case-insensitive - from what I understand from the docs 🙄 
- NFS shares use Posix semantics for file access, and I wanted to avoid a possible small discrepancy between dev and staging/prod